pyqg/tests/test_xarray_output.py

```diff
@@ -4,7 +4,15 @@
 import xarray as xr
 
 def test_xarray():
-    m = pyqg.QGModel(1)
+    # Set parameters & initialize model
+    year = 24*60*60*360.
+    twrite = 1000
+    tavestart = 2*year
+
+    # Initialize QG Model
+    m = pyqg.QGModel(tmax=5*year, twrite=twrite, tavestart=tavestart)
+    m.run()
```

Comment:
Two comments:
- You don't have to run the model for 5 years to test this feature! Just a few timesteps is enough.
- You should make sure to cover both cases: before _and after_ the diagnostics have been initialized. So you probably need `run_with_snapshots`.

Reply:
Using `run_with_snapshots` and testing `m.to_dataset()` before and after diagnostics are calculated. 

https://github.com/hscannell/pyqg/blob/cb1d959589daa7ce8f490ea271637fca3df9390b/pyqg/tests/test_xarray_output.py#L6-L88